src/core/Report.cpp

```diff
@@ -23,48 +35,95 @@
 #include <sstream>
 #include <fstream>
 
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <locale>
+#include <iomanip>
+#include <codecvt>
+
+#include "pugixml/pugixml.hpp"
+
+
+template<class Facet>
+struct deletable_facet : Facet
+{
+    template<class ...Args>
+    deletable_facet(Args&& ...args) : Facet(std::forward<Args>(args)...) {}
+    ~deletable_facet() {}
+};
+
 /*
   It writes a header record and a "Start Report" record.
 */
-void CReport::StartReport(LPCTSTR tcAction, const stringT &csDataBase)
+void CReport::StartReport(LPCTSTR tcAction, const stringT &csDataBase, bool writeHeader)
 {
   m_osxs.str(_T(""));
 
   m_tcAction = tcAction;
   m_csDataBase = csDataBase;
 
-  stringT cs_title, sTimeStamp;
-  PWSUtil::GetTimeStamp(sTimeStamp);
-  Format(cs_title, IDSC_REPORT_TITLE1, tcAction, sTimeStamp.c_str());
-  WriteLine();
-  WriteLine(cs_title);
-  Format(cs_title, IDSC_REPORT_TITLE2, csDataBase.c_str());
-  WriteLine(cs_title);
-  WriteLine();
-  LoadAString(cs_title, IDSC_START_REPORT);
-  WriteLine(cs_title);
-  WriteLine();
+  if(writeHeader) {
+    stringT cs_title, sTimeStamp;
+    PWSUtil::GetTimeStamp(sTimeStamp, true);
+#if !defined(_WIN32) || defined(__WX__)
```

Comment:
Will convert Windows code to match the wx implementation. No good reason for conditional compilation here.

Reply:
Fine with me

Reply:
I got the idea to add a dot in front of the file name of the Report to hide in unix/Mac OS environment. When performing the change please decide if this is a good idea.

Reply:
What purpose would hiding report files serve?
"dot files" are usually configuration files, not generated reports.
Also, the default PasswordSafe directory is ~/.pwsafe, so files there are "hidden" from casual browsing.

Reply:
The folder will be crowded with report files in parallel with password file. As the kind of storage is not relevant and the usr can see inside the application not external visibility is needed, except when moving the password file. When moving the password file the hidden reports would be left at the old location, as normal users would not be aware on its presence. As I told you, it is an idea only to concentrate more on the password file and instead of having a list of reports in-between. Decide by your own what is best, to have the reports visible in the normal file system or having better overview and looking only on password file(s).